topk.go

```diff
@@ -209,30 +220,34 @@ func (s *Stream) Insert(x string, count int) Element {
 	// replace the current minimum element
 	minKey := s.k.elts[0].Key
 
-	mkhash := reduce(sip13.Sum64Str(0, 0, minKey), len(s.alphas))
+	mkhash := reduce(sip13.Sum64Str(0, 0, *minKey), len(s.alphas))
 	s.alphas[mkhash] = s.k.elts[0].Count
 
-	e := Element{
-		Key:   x,
+	e := element{
+		Key:   ptr,
 		Error: s.alphas[xhash],
 		Count: s.alphas[xhash] + count,
 	}
 	s.k.elts[0] = e
 
 	// we're not longer monitoring minKey
-	delete(s.k.m, minKey)
+	delete(s.k.m, *minKey)
 	// but 'x' is as array position 0
 	s.k.m[x] = 0
 
 	heap.Fix(&s.k, 0)
-	return e
+	return Element{Key: *e.Key, Count: e.Count, Error: e.Error}
 }
 
 // Keys returns the current estimates for the most frequent elements
 func (s *Stream) Keys() []Element {
-	elts := append([]Element(nil), s.k.elts...)
+	elts := append([]element(nil), s.k.elts...)
```

Comment:
The `elts` copy seems entirely superfluous... Why not sort the converted array? That should also bring the perforamnce of `s.Keys()` back to comparable with what it was before